dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -826,6 +827,19 @@ private Workflow createOrGetWorkflow(Class workflowType, String repository, User
             workflowToUpdate.getUsers().add(user);
         }
 
+        // Update the manual topic if it's not blank in the .dockstore.yml.
+        // Purposefully not clearing the manual topic when it's null in the .dockstore.yml because another version may have set it
+        if (workflowType != Service.class && wf.getTopic() != null) {
+            if (StringUtils.isNotBlank(wf.getTopic())) {
+                workflowToUpdate.setTopicManual(wf.getTopic());
+                workflowToUpdate.setTopicSelection(TopicSelection.MANUAL);
+            } else {
+                // Clear manual topic if the user purposefully put an empty string
+                workflowToUpdate.setTopicManual(null);
+                workflowToUpdate.setTopicSelection(TopicSelection.AUTOMATIC);
```

Comment:
If the user has selected the AI topic, do we want to leave the selection the way it is?

Reply:
Oh right, I'll have to merge this PR first and modify that in the other PR. Good catch, and yeah I think we would want to leave it as is if the user selected AI